src/sonic-config-engine/minigraph.py

```diff
@@ -572,7 +573,9 @@ def parse_meta(meta, hname):
                     region = value
                 elif name == "CloudType":
                     cloudtype = value
-    return syslog_servers, dhcp_servers, ntp_servers, tacacs_servers, mgmt_routes, erspan_dst, deployment_id, region, cloudtype
+                elif name == "SubRole":
```

Comment:
The SubRole is present only when generating the configuration for individual ASICs in a multi ASIC platform
It is handled seperately in this [function ](https://github.com/Azure/sonic-buildimage/blob/eb76a63b09573ca3adb199fb85a9ac4ad35a8833/src/sonic-config-engine/minigraph.py#L613)

Why do we need introduce SubRole in DEVICE_METADATA for all the devices ?

Reply:
This is so that SubRole can be used for policy enforcement. We will be using it for the internal policies.